src/components/AddTaskForm.tsx

```diff
@@ -0,0 +1,73 @@
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { taskSchema, type TaskFormData } from "../schemas/taskSchema";
+
+type TaskFormProps = {
+  onSubmit: (data: TaskFormData) => void;
+};
+export const TaskForm: React.FC<TaskFormProps> = ({ onSubmit }) => {
```

Comment:
As a general rule I recommend typing components like this:
```suggestion
export const TaskForm = ({ onSubmit }: TaskFormProps) => {
```

And if we have a lot of properties on the props:
```suggestion
export const TaskForm = (props: TaskFormProps) => {
const { prop1, prop2, prop3, prop4 } = props
```